WordPress/src/main/java/org/wordpress/android/ui/mysite/DomainRegistrationHandler.kt

```diff
@@ -1,43 +1,58 @@
 package org.wordpress.android.ui.mysite
 
-import androidx.lifecycle.LiveData
-import androidx.lifecycle.MediatorLiveData
-import androidx.lifecycle.MutableLiveData
+import kotlinx.coroutines.CancellableContinuation
+import kotlinx.coroutines.CancellationException
+import kotlinx.coroutines.flow.flow
+import kotlinx.coroutines.suspendCancellableCoroutine
 import org.greenrobot.eventbus.Subscribe
 import org.greenrobot.eventbus.ThreadMode
 import org.wordpress.android.fluxc.Dispatcher
 import org.wordpress.android.fluxc.generated.SiteActionBuilder
 import org.wordpress.android.fluxc.model.SiteModel
 import org.wordpress.android.fluxc.store.SiteStore.OnPlansFetched
+import org.wordpress.android.fluxc.utils.AppLogWrapper
+import org.wordpress.android.ui.mysite.MySiteUiState.PartialState.DomainCreditAvailable
 import org.wordpress.android.ui.plans.isDomainCreditAvailable
-import org.wordpress.android.util.AppLog
 import org.wordpress.android.util.AppLog.T.DOMAIN_REGISTRATION
 import org.wordpress.android.util.SiteUtilsWrapper
 import javax.inject.Inject
+import javax.inject.Singleton
+import kotlin.coroutines.resume
 
+@Singleton
 class DomainRegistrationHandler
 @Inject constructor(
     private val dispatcher: Dispatcher,
     private val selectedSiteRepository: SelectedSiteRepository,
+    private val appLogWrapper: AppLogWrapper,
     private val siteUtils: SiteUtilsWrapper
-) {
-    private val _sitePlansFetched = MutableLiveData<OnPlansFetched>()
-    val isDomainCreditAvailable: LiveData<Boolean> = MediatorLiveData<Boolean>().apply {
-        addSource(selectedSiteRepository.selectedSiteChange) {
-            it?.let { site ->
-                if (shouldFetchPlans(site)) {
+) : MySiteSource<DomainCreditAvailable> {
+    private var continuation: CancellableContinuation<OnPlansFetched>? = null
+
+    override fun buildSource(siteId: Int) = flow {
+        continuation?.cancel()
+        continuation = null
+        val site = selectedSiteRepository.getSelectedSite()
+        if (site == null || site.id != siteId) {
+            return@flow
+        }
```

Comment:
Same question I had about `ScanAndBackupSource`: I'm not sure why we need to compare the site ids here.

Reply:
we probably don't need that, I just wanted to be super sure that we're loading data for the correct site.